### PR TITLE
add example to run ABOHB on pytorch model

### PR DESCRIPTION
**Pull request by aaronkl**
adds an example python script to show how to optimize the hyperparameters of a pytorch models with ABOHB

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/hnas/abohb_example.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        config["dropout_2"] = dropout_2
+        config["batch_size"] = batch_size
+        config["learning_rate"] = learning_rate
+        config["wd"] = wd
+
+        ts_now = time.time()
+        eval_time = ts_start - ts_now
+        reporter(
+            epoch=epoch + 1,
+            performance=float(curr_best),
```
Comment (mseeger) [inline examples/hnas/abohb_example.py:115]: Strictly speaking, this is not the accuracy after epoch epochs, but the best accuracy over these epochs. Do we want that?

File: examples/hnas/abohb_example.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        # We also report the hyperparameter configuration back to AutoGluon,
+        # so we can process it later in the callback function, for example for visualization.
+        config = dict()
+        config["n_units_1"] = n_units_1
+        config["n_units_2"] = n_units_2
+        config["dropout_1"] = dropout_1
+        config["dropout_2"] = dropout_2
+        config["batch_size"] = batch_size
+        config["learning_rate"] = learning_rate
+        config["wd"] = wd
```
Comment (mseeger) [inline examples/hnas/abohb_example.py:109]: This is not passed back to reporter, right?

File: examples/hnas/abohb_example.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
+import autogluon as ag
```
Comment (mseeger) [inline examples/hnas/abohb_example.py:1]: Very cool. Let us also replace the other examples/tutorials/etc (MXNet code) by PyTorch.

File: examples/hnas/abohb_example.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                                                num_trials=args.num_trials,
+                                                time_out=args.timeout,
+                                                # This argument defines the metric that will be maximized.
+                                                # Make sure that you report this back in the objective function.
+                                                reward_attr='performance',
+                                                # The metric along we make scheduling decision. Needs to be also
+                                                # reported back to AutoGluon in the objective function.
+                                                time_attr='epoch',
+                                                brackets=brackets,
+                                                max_t=27,  # maximum number of epochs to train the network
```
Comment (mseeger) [inline examples/hnas/abohb_example.py:216]: You do not need to set this, it will get it from the train_fn (from args.epochs).

File: examples/hnas/abohb_example.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                                                checkpoint=None,
```
Comment (mseeger) [inline examples/hnas/abohb_example.py:217]: The default is None anyway

**Comment by Innixma** [comment]
To fix the CI, please rebase to mainline. awscli update broke CI recently.

**Comment by szha** [comment]
Job PR-673-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-673/2/index.html

File: examples/hnas/abohb_example.py (hunk 1 above)
Comment (aaronkl) [reply examples/hnas/abohb_example.py:115]: It's a matter of perspective. The intuition is, that, in practice, you wouldn’t trust that the performance at the final epochs is the best. As practitioner you would checkpoint your model after each epoch if the validation performance has improved, and afterwards take the last checkpointed model rather than the final weights.

File: examples/hnas/abohb_example.py (hunk 2 above)
Comment (aaronkl) [reply examples/hnas/abohb_example.py:109]: you are right, I will fix it

**Comment by szha** [comment]
Job PR-673-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-673/3/index.html

**Comment by Innixma** [comment]
@aaronkl Are you looking for reviews from anyone in particular at this time, or are you working on this with Matthias still?

**Comment by aaronkl** [comment]
I fixed all of Matthias comments, so if he is fine with it we can merge it

**Comment by Innixma** [comment]
@aaronkl Why does this PR show that it is adding so many lines? Most of these changes appear to be changes made by Jonas that were already merged into mainline. Do you know why this is the case?

**Comment by aaronkl** [comment]
@Innixma  I have no idea to be honest, I guess something went wrong with the rebase. Do you want me to create a new PR to fix this?

**Comment by Innixma** [comment]
@aaronkl Yeah, I think it would be best to make a new PR just to clean up the changes. It is definitely weird, not sure why it looks like that.

**Comment by aaronkl** [comment]
ok new PR is now #687
